cli/src/main/java/com/devonfw/tools/ide/util/FilenameUtil.java

```diff
@@ -28,6 +28,12 @@ public static String getExtension(String path) {
       lastSlash = 0;
     }
     int lastDot = path.lastIndexOf('.');
+
+    // workaround for sourceforge urls
+    if (path.contains("sourceforge") && path.endsWith("download")) {
+      return path.substring(lastDot + 1, lastSlash);
+    }
```

Comment:
I am not sure, if I fully got this but it seems we are using this method with an URL instead of a `path`.
It was not really designed for this but if we have a sourceforge URL and the actual download filename is somehow a folder in the path I am fine with this workaround. Thanks for adding the comment to make it more clear. Probably add an example URL to make it even more obvious what is going on here.

Reply:
Sorry for perfection but we could additionally check that path starts with `https://` to make this clear and avoid errors in strange edge cases.

Reply:
I have added the check with `startsWith(https://)`. We could also take `https://sourceforge` as expression, but I don't if there are subdomains that would fail the check, so I keep `sourceforge` as separate expression.